### PR TITLE
Added code editor to injection interface: the sequel (this time, with CodeMirror!)

### PR DESCRIPTION
**Pull request by acburdine**
This adds code syntax highlighting to the code-injection interface

references #4995

### CODE REVIEW

**Comment by ErisDS** [comment]
Whoop, filesize is now 2.4 instead of 2.6 :) Screenshot of how this looks:



cc @JohnONolan

**Comment by JohnONolan** [comment]
Could we just use the Github-style theme please?

**Comment by acburdine** [comment]
@JohnONolan there isn't a github theme for codemirror. Do you want me to create one?

**Comment by ErisDS** [comment]
I think the [neat](https://codemirror.net/demo/theme.html) theme is pretty close?

**Comment by JohnONolan** [comment]
xq-light would be ideal - preferably with the "current line highlight" disabled

**Comment by acburdine** [comment]
Done. I didn't see any current line highlight option in Codemirror, but after applying it and looking at the component itself, it doesn't look like that theme actually highlights the currently selected line, so it should be good.

File: core/client/app/components/gh-cm-editor.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    editor: null, // reference to CodeMirror editor
+
+    // options for the editor
+    lineNumbers: true,
+    indentUnit: 4,
+    mode: 'htmlmixed',
+    theme: 'xq-light',
+
+    didInsertElement: function () {
+        var options = this.getProperties('lineNumbers', 'indentUnit', 'mode', 'theme'),
```
Comment (novaugust) [inline core/client/app/components/gh-cm-editor.js]: <3 :D

**Comment by ErisDS** [comment]
Nope :grin:

**Pull request merged**